packages/components/src/dropdown/index.js

```diff
@@ -81,17 +80,30 @@ export default function Dropdown( {
 
 	const args = { isOpen, onToggle: toggle, onClose: close };
 
+	const isPressingToggle = useRef();
+	let toggleElement = renderToggle( args );
+	const { props } = toggleElement;
+	const pressHandlers = {
+		onMouseDown: ( event ) => {
+			isPressingToggle.current = true;
+			props.onMouseDown?.( event );
+		},
+		onMouseUp: ( event ) => {
+			isPressingToggle.current = false;
+			props.onMouseUp?.( event );
+		},
+	};
+	toggleElement = cloneElement( toggleElement, pressHandlers );
```

Comment:
Clever solutions, Unfortunately, this is a breaking change in WordPress and something  we can't ship just like that. Potentially we can try to detect whether we receive just a single element. 

The problem though is even if it's just a single element, there's no guarantee that this element supports these props. So for me instead of doing this like that, we could potentially add these two handlers as new ` args` to the render callback and have the consumer add them.

Reply:
>there's no guarantee that this element supports these props
True, one of the adjustments made in this PR was a to `ButtonBlockAppender` which didn't work even though it already returned a single element.

>we could potentially add these two handlers as new args to the render callback and have the consumer add them
I'm happy to polish up my other PR #30397 which does just that.

Thank you for reviewing 🙇